process/transaction/baseProcess.go

```diff
@@ -0,0 +1,144 @@
+	accounts         state.AccountsAdapter
+	shardCoordinator sharding.Coordinator
+	adrConv          state.AddressConverter
+}
+
+func (txProc *baseTxProcessor) getAccounts(
+	adrSrc, adrDst state.AddressContainer,
+) (*state.Account, *state.Account, error) {
+
+	var acntSrc, acntDst *state.Account
+
+	shardForCurrentNode := txProc.shardCoordinator.SelfId()
+	shardForSrc := txProc.shardCoordinator.ComputeId(adrSrc)
+	shardForDst := txProc.shardCoordinator.ComputeId(adrDst)
+
+	srcInShard := shardForSrc == shardForCurrentNode
+	dstInShard := shardForDst == shardForCurrentNode
+
+	if srcInShard && adrSrc == nil ||
+		dstInShard && adrDst == nil {
+		return nil, nil, process.ErrNilAddressContainer
+	}
+
+	if bytes.Equal(adrSrc.Bytes(), adrDst.Bytes()) {
+		acntWrp, err := txProc.accounts.GetAccountWithJournal(adrSrc)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		account, ok := acntWrp.(*state.Account)
+		if !ok {
+			return nil, nil, process.ErrWrongTypeAssertion
+		}
+
+		return account, account, nil
+	}
+
+	if srcInShard {
+		acntSrcWrp, err := txProc.accounts.GetAccountWithJournal(adrSrc)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		account, ok := acntSrcWrp.(*state.Account)
+		if !ok {
+			return nil, nil, process.ErrWrongTypeAssertion
+		}
+
+		acntSrc = account
+	}
+
+	if dstInShard {
+		acntDstWrp, err := txProc.accounts.GetAccountWithJournal(adrDst)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		account, ok := acntDstWrp.(*state.Account)
+		if !ok {
+			return nil, nil, process.ErrWrongTypeAssertion
+		}
+
+		acntDst = account
+	}
+
+	return acntSrc, acntDst, nil
+}
+
+func (txProc *baseTxProcessor) getAccountFromAddress(adrSrc state.AddressContainer) (state.AccountHandler, error) {
+	shardForCurrentNode := txProc.shardCoordinator.SelfId()
+	shardForSrc := txProc.shardCoordinator.ComputeId(adrSrc)
+	if shardForCurrentNode != shardForSrc {
+		return nil, nil
+	}
+
+	acnt, err := txProc.accounts.GetAccountWithJournal(adrSrc)
+	if err != nil {
+		return nil, err
+	}
+
+	return acnt, nil
+}
+
+func (txProc *baseTxProcessor) getAddresses(
+	tx *transaction.Transaction,
+) (state.AddressContainer, state.AddressContainer, error) {
+	adrSrc, err := txProc.adrConv.CreateAddressFromPublicKeyBytes(tx.SndAddr)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	adrDst, err := txProc.adrConv.CreateAddressFromPublicKeyBytes(tx.RcvAddr)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return adrSrc, adrDst, nil
+}
+
+func (txProc *baseTxProcessor) checkTxValues(tx *transaction.Transaction, acntSnd state.AccountHandler) error {
+	if acntSnd == nil || acntSnd.IsInterfaceNil() {
+		// transaction was already done at sender shard
+		return nil
+	}
+
+	if acntSnd.GetNonce() < tx.Nonce {
+		return process.ErrHigherNonceInTransaction
+	}
+	if acntSnd.GetNonce() > tx.Nonce {
+		return process.ErrLowerNonceInTransaction
+	}
+
+	cost := big.NewInt(0)
+	cost = cost.Mul(big.NewInt(0).SetUint64(tx.GasPrice), big.NewInt(0).SetUint64(tx.GasLimit))
```

Comment:
You could use `cost.Mul(big.NewInt(0).SetUint64(tx.GasPrice), big.NewInt(0).SetUint64(tx.GasLimit))` without `cost = `. Also for the next line.

Reply:
done